### ISSUE TITLE
Kodi 18 Leia Python3 compatibility

### ISSUE THREAD

**Issue by bigretromike** [bigretromike/nakamori#293]
Here are tools for test anything that should be compatible with Python3 Kodi interpreter:
- [Windows 32-bit](http://mirrors.xbmc.org/test-builds/windows/win32/)
- [Windows 64-bit](http://mirrors.kodi.tv/test-builds/windows/win64/)
- [Linux/PPA](https://launchpad.net/~wsnipex/+archive/ubuntu/kodi-python3/)

**Comment by bigretromike** [comment]
I pushed https://github.com/bigretromike/nakamori/commit/384a171262e3b42e9e039083ea42f1e9a699a5a1 that enable us to load proper libs from python3, sadly there is more issue than that string <-> unicode errors in nakamoripluign.py build. Old one works fine (almost but it looks new git push broken PIL)

**Comment by bigretromike** [comment]
Python PIL(LOW) Python3 issue: https://forum.kodi.tv/showthread.php?tid=341419

**Comment by bigretromike** [comment]
- [ ] `NameError: name 'unicode' is not defined`
- [ ] `script.module.nakamori-lib\lib\nakamori_utils\shoko_utils.py", line 329, in can_connect`
- [ ] `script.module.nakamori-lib\lib\class_dump.py", line 16, in dump_to_text`

**Comment by bigretromike** [comment]
- [ ] `AttributeError: 'RuntimeError' object has no attribute 'message'`
- [ ] `script.module.nakamori-lib\lib\error_handler.py", line 231, in exception_internal`

**Comment by bigretromike** [comment]
- [ ] `2019-03-14 13:20:55.993 T:12356   ERROR: Nakamori|Logger -> There was an error while executing Nakamori.
2019-03-14 13:20:55.994 T:12356   ERROR: Nakamori|Logger -> Nakamori Version 3.1.0~beta2
2019-03-14 13:20:55.994 T:12356   ERROR: Nakamori|Logger -> The url accessed was plugin://plugin.video.nakamori/
2019-03-14 13:20:55.994 T:12356   ERROR: Nakamori|Logger -> Cannot connect to Shoko Server. Please check your ip settings. -- Exception: RuntimeError at ./plugin.video.nakamori/nakamoriplugin.py#L238
2019-03-14 13:20:55.994 T:12356   ERROR: Nakamori|Logger -> Traceback (most recent call last):
2019-03-14 13:20:55.994 T:12356   ERROR: Nakamori|Logger ->   File "./plugin.video.nakamori/nakamoriplugin.py", line 224, in _main
2019-03-14 13:20:55.994 T:12356   ERROR: Nakamori|Logger ->     raise RuntimeError('Cannot connect to Shoko Server. Please check your ip settings.')
2019-03-14 13:20:55.994 T:12356   ERROR: Nakamori|Logger -> RuntimeError: Cannot connect to Shoko Server. Please check your ip settings.
2019-03-14 13:20:57.420 T:12356   ERROR: CPythonInvoker(12, \Roaming\Kodi\addons\plugin.video.nakamori\nakamoriplugin.py): failed to set abortRequested
2019-03-14 13:20:57.447 T:12412   ERROR: XFILE::CDirectory::GetDirectory - Error getting plugin://plugin.video.nakamori/
2019-03-14 13:20:57.457 T:7832   ERROR: CGUIMediaWindow::GetDirectory(plugin://plugin.video.nakamori/) failed`

**Comment by da3dsoul** [comment]
Last one isn't Leia. Also damnit it didn't warn me about that. Mk

**Comment by da3dsoul** [comment]
I didn't really care that much about this, but apparently Python3's json is 3x faster, so that's kind of a big deal for us.

**Comment by bigretromike** [comment]
> 
> 
> I didn't really care that much about this, but apparently Python3's json is 3x faster, so that's kind of a big deal for us.

is it still worth to replace it with protobuf ?

**Comment by da3dsoul** [comment]
Yes, but json being fast is most relevant when calling Kodi's json RPC and Shoko's API, where protobuf aren't supported

**Comment by bigretromike** [comment]
Looks like Team-Kodi stoped building python3 builds.
From what I remembered they accounced that v19 (Matrix) will have python3 build - but for a reason there is still non p3 builds od v19. 

We had a good record with python3 compatibility across last python3 buillds, so I close this and will open v19 p3 issue when that will be a thing - without PIL it was utter hard to test everything without commenting out sections of code.

We did a great job with this.
